src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/service/RecallNotificationServiceTest.kt

```diff
@@ -36,32 +41,46 @@ internal class RecallNotificationServiceTest {
   @Test
   fun `get recall notification returns recall summary, revocation order and letter to probation when one doesnt exist already`() {
     val recallId = ::RecallId.random()
-    val recallSummaryBytes = randomString().toByteArray()
-    val revocationOrderBytes = randomString().toByteArray()
-    val letterToProbationBytes = randomString().toByteArray()
+    val recallSummaryContent = randomString()
+    val revocationOrderContent = randomString()
+    val letterToProbationContent = randomString()
     val mergedBytes = randomString().toByteArray()
     val documentId = UUID.randomUUID()
     val userId = UserId(UUID.randomUUID())
+    val documentsToMergeSlot = slot<List<InputStreamDocumentData>>()
 
-    every { recallDocumentService.getDocumentContentWithCategoryIfExists(recallId, RecallDocumentCategory.RECALL_NOTIFICATION) } returns null
-    every { recallSummaryService.getPdf(recallId) } returns Mono.just(recallSummaryBytes)
-    every { revocationOrderService.createPdf(recallId, userId) } returns Mono.just(revocationOrderBytes)
-    every { letterToProbationService.getPdf(recallId) } returns Mono.just(letterToProbationBytes)
-    every { pdfDocumentGenerationService.mergePdfs(any()) } returns Mono.just(mergedBytes) // TODO: test that the argument list has 3 entries; the correct entries; but note likely to change call to decouple from Gotenberg
-    every { recallDocumentService.uploadAndAddDocumentForRecall(recallId, mergedBytes, RecallDocumentCategory.RECALL_NOTIFICATION) } returns documentId
+    every { recallDocumentService.getDocumentContentWithCategoryIfExists(recallId, RECALL_NOTIFICATION) } returns null
+    every { recallSummaryService.getPdf(recallId) } returns Mono.just(recallSummaryContent.toByteArray())
+    every { revocationOrderService.createPdf(recallId, userId) } returns Mono.just(revocationOrderContent.toByteArray())
+    every { letterToProbationService.getPdf(recallId) } returns Mono.just(letterToProbationContent.toByteArray())
+    every { pdfDocumentGenerationService.mergePdfs(capture(documentsToMergeSlot)) } returns Mono.just(mergedBytes)
+    every { recallDocumentService.uploadAndAddDocumentForRecall(recallId, mergedBytes, RECALL_NOTIFICATION) } returns documentId
 
-    val result = underTest.getDocument(recallId, userId)
+    val recallNotification = underTest.getDocument(recallId, userId).block()!!
 
-    StepVerifier.create(result).assertNext { assertThat(it, equalTo(mergedBytes)) }.verifyComplete()
+    assertThat(recallNotification, equalTo(mergedBytes))
+    assertThat(
+      documentsToMergeSlot.captured,
+      onlyContainsInOrder(
+        listOf(
+          documentWith(recallSummaryContent),
+          documentWith(revocationOrderContent),
+          documentWith(letterToProbationContent)
+        )
+      )
+    )
   }
 
+  private fun documentWith(documentContent: String) =
```

Comment:
Two copies of this: just wondering what the "Kotlinic"? way to share them might be?

Reply:
Yes I was lazy here, thought about reusing it but went for the cut and paste option.  I'll move somewhere when I next come across it